packages/nimble-components/build/generate-workers/source/matrix-renderer.ts

```diff
@@ -187,5 +232,59 @@ export class MatrixRenderer {
             }
         }
     }
+
+    public drawText(): void {
+        this.context.font = `${this.renderConfig.labelsFontSize.toString()}px sans-serif`;
+        this.context.fillStyle = '#ffffff';
+        this.context.textAlign = 'center';
+        this.context.lineCap = 'butt';
+        const approximateTextHeight = this.context.measureText('M');
+
+        for (let i = 0; i < this.scaledColumnIndices.length; i++) {
+            const scaledX = this.scaledColumnIndices[i]!;
+            if (
+                !(
+                    scaledX >= this.transformConfig.topLeftCanvasCorner.x
+                    && scaledX < this.transformConfig.bottomRightCanvasCorner.x
+                )
+            ) {
+                continue;
+            }
+
+            // columnIndexPositions is used to get chunks to determine the start and end index of the column, it looks something like [0, 1, 4, 9, 12]
+            // This means that the first column has a start index of 0 and an end index of 1, the second column has a start index of 1 and an end index of 4, and so on
+            // scaledRowIndices is used when we reach the end of the columnIndexPositions, when columnIndexPositions is [0, 1, 4, 9, 12], scaledRowIndices is 13
+            const columnEndIndex = this.columnIndicesPositions[i + 1] !== undefined
+                ? this.columnIndicesPositions[i + 1]!
+                : this.scaledRowIndices.length;
+            for (
+                let columnStartIndex = this.columnIndicesPositions[i]!;
+                columnStartIndex < columnEndIndex;
+                columnStartIndex++
+            ) {
+                const scaledY = this.scaledRowIndices[columnStartIndex]!;
+                if (
+                    !(
+                        scaledY >= this.transformConfig.topLeftCanvasCorner.y
+                        && scaledY < this.transformConfig.bottomRightCanvasCorner.y
+                    )
+                ) {
+                    continue;
+                }
+                let label = `${this.values[columnStartIndex] || 'NaN'}${this.renderConfig.dieLabelsSuffix}`;
```

Comment:
@munteannatan after enabling type checked linting it found a case here where theoretically `this.values[columnStartIndex]` could be `undefined`. I added the `|| 'NaN'` condition but not completely sure if that's how you want it to be handled.

Reply:
@munteannatan I'm going to merge this PR in order to do some follow-up changes, but please consider what's the best thing to do here and handle in a follow-up, etc.

Reply:
Thank you! I will try and check the behavior and make adjustments in a follow up.